src/index.ts

```diff
@@ -0,0 +1,122 @@
+declare function postMessage(data: any);
+
+function runCode(): Point[] {
+    console.log('runcode');
+    const mersenneTwister = new MersenneTwister();
+    let minimumReached = false;
+
+    let seekingMode = new ClassicalSeekingMode(
+        config.seekingMemoryPool,
+        config.seekingRangeOfSelectedDimension,
+        config.countsOfDimensionsToChange,
+        config.selfPositionConsidering);
+    let tracingMode = new ClassicalTracingMode(config.constantNumber, searchDomain);
+    let cats: Cat[] = [];
+    let lastResult: Position;
+    let bestPosition: Position;
+    let lastTwentyCats: number[] = [];
+    let results: Position[] = [];
+    let fitnessValueBest = Number.POSITIVE_INFINITY;
+    let fitnessValueLeast = Number.NEGATIVE_INFINITY;
+    const minimumEpsilon = 1e-5;
+
+    for (let i = 0; i < config.numberOfCats; i++) {
+        cats[i] = new Cat(
+            Position.doRandomPosition(
+                searchDomain.min.x,
+                searchDomain.max.x,
+                searchDomain.min.y,
+                searchDomain.max.y),
+            Position.doRandomPosition(),
+            functionToOptimize);
+    }
+    for (let iterationCounter = 0;
+        !minimumReached &&
+        iterationCounter < config.maximumNumberOfIterations;
+        iterationCounter++) {
+        for (let cat of cats) {
+            const fitness = cat.calculateFitness();
+            if (fitness < fitnessValueBest) {
+                fitnessValueBest = fitness;
+                bestPosition = cat.Position;
+            }
+            if (fitness > fitnessValueLeast) {
+                fitnessValueLeast = fitness;
+            }
+            cat.Mode = mersenneTwister.random() >= config.mixtureRatio;
+        }
+        /*  console.log(cats);
+          break;*/
+        if (bestPosition !== undefined &&
+            (lastResult === undefined ||
+                !lastResult.compareToPoint(bestPosition, minimumEpsilon))) {
+            lastResult = bestPosition;
+            results.push(lastResult);
+        }
+        else {
+            console.log([
+                lastResult, bestPosition,
+                fitnessValueBest, fitnessValueLeast,
+                lastResult && bestPosition && lastResult.subtract(bestPosition.x, bestPosition.y),
+                lastResult && bestPosition && !lastResult.compareToPoint(bestPosition, minimumEpsilon)]
+            );
+        }
+        for (let i = 0; i < config.numberOfCats; i++) {
+            if (cats[i].Mode) {
+                seekingMode.seek(cats[i], fitnessValueBest, fitnessValueLeast);
+                break;
+            } else {
+                if (bestPosition === undefined) {
+                    console.log('no bestposition!');
+                }
+                tracingMode.trace(cats[i], bestPosition);
+            }
+        }
+        lastTwentyCats[iterationCounter % 20] = fitnessValueBest;
+        if (iterationCounter > 19) {

```

Comment:
why 19? what is this magic number doing?

Reply:
put this absolute number into parameters.json and config, so it is a parameter now